tests/test_sft_trainer.py

```diff
@@ -0,0 +1,325 @@
+
+
+############################# Prompt Tuning Tests #############################
+
+
+def test_run_causallm_pt_and_inference():
+    """Check if we can bootstrap and peft tune causallm models"""
+    with tempfile.TemporaryDirectory() as tempdir:
+        BASE_PEFT_KWARGS["output_dir"] = tempdir
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            BASE_PEFT_KWARGS
+        )
+        sft_trainer.train(model_args, data_args, training_args, tune_config)
+
+        # validate peft tuning configs
+        _validate_training(tempdir)
+        checkpoint_path = _get_checkpoint_path(tempdir)
+        adapter_config = _get_adapter_config(checkpoint_path)
+        _validate_adapter_config(adapter_config, "PROMPT_TUNING", BASE_PEFT_KWARGS)
+
+        # Load the model
+        loaded_model = TunedCausalLM.load(checkpoint_path)
+
+        # Run inference on the text
+        output_inference = loaded_model.run(
+            "### Text: @NortonSupport Thanks much.\n\n### Label:", max_new_tokens=50
+        )
+        assert len(output_inference) > 0
+        assert "### Text: @NortonSupport Thanks much.\n\n### Label:" in output_inference
+
+
+def test_run_causallm_pt_init_text():
+    """Check if we can bootstrap and peft tune causallm models with init text as 'TEXT'"""
+    with tempfile.TemporaryDirectory() as tempdir:
+        pt_init_text = copy.deepcopy(BASE_PEFT_KWARGS)
+        pt_init_text["output_dir"] = tempdir
+        pt_init_text["prompt_tuning_init"] = "TEXT"
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            pt_init_text
+        )
+        sft_trainer.train(model_args, data_args, training_args, tune_config)
+
+        # validate peft tuning configs
+        _validate_training(tempdir)
+        checkpoint_path = _get_checkpoint_path(tempdir)
+        adapter_config = _get_adapter_config(checkpoint_path)
+        _validate_adapter_config(adapter_config, "PROMPT_TUNING", pt_init_text)
+
+
+invalid_params_map = [
+    ("num_train_epochs", 0, "num_train_epochs has to be an integer/float >= 1"),
+    (
+        "gradient_accumulation_steps",
+        0,
+        "gradient_accumulation_steps has to be an integer >= 1",
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "param_name,param_val,exc_msg",
+    invalid_params_map,
+    ids=["num_train_epochs", "grad_acc_steps"],
+)
+def test_run_causallm_pt_invalid_params(param_name, param_val, exc_msg):
+    """Check if error is raised when invalid params are used to peft tune causallm models"""
+    with tempfile.TemporaryDirectory() as tempdir:
+        invalid_params = copy.deepcopy(BASE_PEFT_KWARGS)
+        invalid_params["output_dir"] = tempdir
+        invalid_params[param_name] = param_val
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            invalid_params
+        )
+
+        with pytest.raises(ValueError, match=exc_msg):
```

Comment:
I generally avoid matching exact error message and just check for valueError with a comment explaining why, but will let this go and I dont think we will update the message much